proposals/xxxx-mesh-nodes.md

```diff
@@ -0,0 +1,329 @@
+[NodeLaunch("mesh")]
+[NodeIsProgramEntry]
+[NodeID("meshNode", 2)]
+[NodeLocalRootArgumentsTableIndex(4)]
+[NumThreads(4,4,1)]
+[NodeMaxDispatchGrid(16,16,1)]
+[OutputTopology("triangle")]
+void MyMeshNode(
+    DispatchNodeInputRecord<MyMeshRecord> nodeInput,
+    out vertices PerVertexValues verts[16],
+    out primitives PerPrimitiveValues prim[8],
+    out indices uint3 tris[16] )
+{
+  float3 data = nodeInput.Get().perMeshConstant;
+  // compute numVerts, numPrims...
+  SetMeshOutputCounts(numVerts, numPrims);
+  // Cooperate across group to write output arrays...
+}
+```
+
+### Compiling Experimental Mesh Node Shaders
+
+As a kind of node shader in work graphs, mesh node shaders are functions
+defined in a library.
+To use the experimental feature, you must compile HLSL with options:
+`-T lib_6_9 -select-validator internal`.
+This produces a DXIL library for use with the state object API in accordance
+with the work graphs spec
+[here](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#graphics-nodes.
+
+`-select-validator internal` is for use during feature development to avoid
+using an external released validator that does not recognize the feature or
+the in-development shader model being used.
+
+## Detailed design
+
+### HLSL Additions
+
+A new `[NodeLaunch("mesh")]` mode for `[shader("node")]` indicates a mesh node
+shader entry point.  This section describes this new entry type.
+
+#### Function Attributes
+
+Supports attributes from both broadcasting launch node and mesh shader, using
+`[NodeLaunch("mesh")]`. The following table is strictly from the perspective of
+mesh node shaders. Other values, or different descriptions could apply for
+other node or shader types.
+See [Shader function attributes in Work Graphs spec](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#shader-function-attributes)
+for more detailed descriptions of the node function attributes.
+
+| Attribute | Required | Description |
+|--|--|--|
+| `[shader("node")]` | Y | Indicates a node shader entry point |
+| `[NodeLaunch("mesh")]` | Y | Signifies a mesh node |
+| `[NumThreads(x,y,z)]` | Y | Specifies the launch size of the threadgroup of the Mesh shader, just like with compute shader. The number of threads can not exceed `X * Y * Z = 128`. See [numthreads](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#numthreads) in the Mesh Shader spec. |
+| `[OutputTopology(`*topology*`)]` | Y | Specifies the topology for the output primitives of the mesh shader. *topology* must be `"line"` or "`triangle"`. See [outputtopology](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/MeshShader.md#outputtopology) in the Mesh Shader spec. |
+| `[NodeDispatchGrid(x,y,z)]` | Yes, unless `NodeMaxDispatchGrid` is defined | Declare a fixed dispatch grid size for use with this node. ([details](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#shader-function-attributes)) |
```

Comment:
Nit: Since this is widening the column anyway, the rest might as well be "Yes" and "No" instead of just "Y" and "N"